buildlib/pr/main.yml

```diff
@@ -173,3 +176,17 @@ stages:
         parameters:
             name: gpu
             demands: ucx_gpu -equals yes
+
+  - stage: python
+    dependsOn: [Static_check]
+    jobs:
+      - template: python/python-test.yml
+        parameters:
+          name: new
+          demands: ucx_new -equals yes
+          container: conda
+      - template: python/python-test.yml
+        parameters:
+          name: gpu
+          demands: ucx_gpu -equals yes
+          container: conda
```

Comment:
@pentschev for this PR, please run only with this:
`demands: ucx_docker -equals yes`
otherwise, you would either run on a machine that does not run a docker daemon (ucx_new) or on a machine that already runs ucx bare-metal (ucx_gpu). Currently , until we fix the issue, running a docker test on the same machine as the bare metal test will cause some bare metal tests to fail because of using the temporary docker bridge interface.

Reply:
I'll update it. Does `ucx_docker` have GPU(s)?

Reply:
unfortunately no, but should still be able to run ucx-py right?
let's add gpu dimension in next or after fixing the above issue

Reply:
Barely, I think, I don't recall us every running on non-GPU machines.. Our main focus is indeed GPUs, so for a first pass it's ok, but we really need GPU support for any reasonable UCX-Py coverage.